packages/compass/src/main/setup-preferences.ts

```diff
@@ -0,0 +1,27 @@
+import Preferences from 'compass-preferences-model';
+import { ipcMain } from 'hadron-ipc';
+
+import { getStoragePaths } from '@mongodb-js/compass-utils';
+const { basepath } = getStoragePaths() || {};
+
+const setupPreferences = async() => {
+  const preferences = new Preferences(basepath);
+
+  await preferences.fetchPreferences();
+
+  ipcMain.handle('compass:save-preferences', async (event, attributes) => {
+    const savedPreferencesValues = await preferences.savePreferences(attributes);
+    ipcMain.broadcast('compass:preferences-changed', savedPreferencesValues);
```

Comment:
Noticed when we were talking that we only emit this event here … ideally we’d also account for changes to the preferences coming from the main process and emit the event in that case. Maybe we could add `onPreferencesChanged` to the `Preferences` class as well, and emit it when `savePreferences()` is called, then register a handler here that does the IPC broadcast?

Reply:
i think in design we agreed that global/cli prefs are set only on startup, what case could be that the preferences change happens in main and not via the 'compass:save-preferences' event?

Reply:
I don’t have a concrete answer, and I agree that we’re currently not planning to introduce anything where this would make a difference, it just feels like a small footgun for future changes that preferences can be modified without triggering this event.